lib/msal-common/src/error/ClientAuthError.ts

```diff
@@ -68,6 +68,10 @@ export const ClientAuthErrorMessages = {
         "Client assertion must meet requirements described in https://tools.ietf.org/html/rfc7515",
     [ClientAuthErrorCodes.invalidClientCredential]:
         "Client credential (secret, certificate, or assertion) must not be empty when creating a confidential client. An application should at most have one credential",
+    [ClientAuthErrorCodes.invalidThumbprint]:
```

Comment:
I think you should allow this. If authority is AAD, prefer SHA2. If authority is ADFS, prefer SHA1.